### DISCUSSION TITLE
Can I bring back dismissed composing text?

### DISCUSSION THREAD

**Discussion by HubKing** [google/mozc#673] [Q&A] [answered]
When  I am typing Japanese, if I dismiss it before pressing the enter key by a mistake, the underlined text is gone. Other than retyping the text again, is there a way to "undo" the dismissal and bring the disappeared text back?

[gone.webm](https://user-images.githubusercontent.com/10900489/212447326-94a92d2f-862a-46cf-a5b8-ab52c20601da.webm)

**Comment by hiroyuki-komatsu** [comment, marked as answer]
Hi HubKing,

Did you dismiss it by pressing the ESC key? IIRC, there is no way to undo at this moment.
If you commit it by pressing Enter key, you can undo it by pressing Ctrl+BackSpace though.

It's indeed an interesting point to consider supporting it.
Thank you,

**Reply by HubKing** [reply]
The reason of dismissal can vary: It can be an excess ESC when I tried to cancel a conversion, or it can be bad palm rejection. On my laptop, the left side of the J key is almost at the centre of the touchpad, so it is extremely easy for my right palm to touch the touchpad by a mistake whilst  I am trying to type something. Unfortunately, Gnome's palm rejection seems worse than that of Windows. I use "tap to click", so an erroneous palm touch could be interpreted as clicking somewhere else, which dismisses the text in composition.
